### PR TITLE
feat: abort-controllers

### LINKED ISSUE #22: feat: request aborting signal
## This would solve...

I think we need a signal. (for request aborting)

## The implementation should look like...

It should be something like putting `AbortController` at option.

## I have also considered...

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by tbnritzdoge**

### CODE REVIEW

File: src/lib/PetitioRequest.ts [hunk 1]
```diff
@@ -122,6 +127,15 @@ export class PetitioRequest {
 
 		return this;
 	}
+	/**
+	 * @param {AbortController} controller A AbortController to potentially abort the request.
```
Comment (lveneris) [inline src/lib/PetitioRequest.ts]:
```suggestion
	 * @param {AbortController} controller A controller instance that handles aborting the request.
```

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Can you also add a usage example for how one would abort the request? I can see people asking about that in the future.

File: src/lib/PetitioRequest.ts [hunk 2]
```diff
@@ -129,6 +134,18 @@ export class PetitioRequest {
 
 		return this;
 	}
+	/**
+	 * @param {AbortController} controller A controller instance that handles aborting the request.
+	 * @return {*} The request object for further composition.
+	 * @example Initiate a AbortController class and then attach it to the request object
+	 * by doing `.signal(the_controller)`. Then you can abort the request by running
+	 * the abort() function on the controller you created.
```
Comment (lveneris) [inline src/lib/PetitioRequest.ts]:
```suggestion
	 * @example
	 * ```ts
	 * const controller = new AbortController();
	 * const result = petitio(URL).signal(controller);
	 * setTimeout(() => controller.abort(), 5000) // serves as a request timeout
	 * ```
	 */
```

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Just a better example writeup.

**Review by Nytelife26 — APPROVED** [review]

**Pull request merged**